### PR TITLE
Segment telemetry

### PR DESCRIPTION
**Pull request by IvanPleshkov**
### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [x] Does your submission pass tests?
2. [x] Have you lint your code locally using ``cargo fmt`` command prior to submission?
3. [x] Have you checked your code using ```cargo clippy``` command?

### Changes to Core Features:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your core changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: lib/segment/src/telemetry.rs [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    pub large_cardinality_searches: TelemetryOperationStatistics,
+    pub positive_check_cardinality_searches: TelemetryOperationStatistics,
+    pub negative_check_cardinality_searches: TelemetryOperationStatistics,
+}
+
+#[derive(Serialize, Clone, Default)]
+pub struct TelemetryOperationStatistics {
+    pub ok_count: usize,
+    pub fail_count: usize,
+    pub ok_avg_time: Duration,
```
Comment (agourlay) [inline lib/segment/src/telemetry.rs:40]: What do you think about using an histogram to record the distribution of latency.
This way we will know about really high max values and weird distribution.

Comment (IvanPleshkov) [reply lib/segment/src/telemetry.rs:40]: Maybe later. For now, we are using sliding window average and I think we gonna change it later

Comment (IvanPleshkov) [reply lib/segment/src/telemetry.rs:40]: We don't have an exact vision of measurements yet

File: lib/collection/src/shard/remote_shard.rs [hunk 2]
```diff
@@ -1,5 +1,5 @@
 use std::path::{Path, PathBuf};
-use std::sync::Arc;
+use std::sync::{Arc, Mutex};
```
Comment (generall) [inline lib/collection/src/shard/remote_shard.rs:2]: also parking lot?

File: lib/segment/src/index/hnsw_index/hnsw.rs [hunk 3]
```diff
@@ -2,7 +2,7 @@ use std::cmp::max;
 use std::fs::create_dir_all;
 use std::path::{Path, PathBuf};
 use std::sync::atomic::{AtomicBool, Ordering};
-use std::sync::Arc;
+use std::sync::{Arc, Mutex};
```
[the remark this answers is not in the document]
Comment (IvanPleshkov) [reply lib/segment/src/index/hnsw_index/hnsw.rs]: Fixed

File: lib/collection/src/shard/remote_shard.rs (hunk 2 above)
Comment (IvanPleshkov) [reply lib/collection/src/shard/remote_shard.rs]: Fixed

**Review by generall — APPROVED** [review]

**Pull request merged**